src/compiler/checker.ts

```diff
@@ -39998,7 +39998,7 @@ namespace ts {
 
         function isAliasResolvedToValue(symbol: Symbol): boolean {
             const target = resolveAlias(symbol);

```

Comment:
I'd recommend this instead just to ensure we evaluate the conditions at the bottom of the function:
```suggestion
            const target = getExportSymbolOfValueSymbolIfExported(resolveAlias(symbol));
```

This will unwrap the `ExportValue` symbol to the underlying symbol

Reply:
Thanks!

Reply:
I tried to use that helper, however, there were two tests that failed. So I decided to use `ExportValue` instead. Especially I worry about that case - `https://github.com/microsoft/TypeScript/pull/44718/files#diff-d988b2b521a3b21ff40a4a21627ee0647acb2bbb0fe987d90a787c7e8b661cc4R12-R13

Reply:
@rbuckton What do you think about the cases that I pointed above?

Reply:
@rbuckton What do you think about the cases that I pointed above?

Reply:
In what way did they fail?

Reply:
Tests fail with using ‘getExportSymbolOfValueSymbolIfExported‘ instead of ExportValue

Reply:
Both 1 and 2 seem correct to me.

Reply:
Ok. Thanks for your review. Do you have any other remarks that I need to fix?

Reply:
Actually, to clarify: 2 looks correct (the source uses `export =` so it shouldn't have `__esModule` in the emit). 1 looks strange, but we're also contending with errors in the checker due to duplicate conflicting declarations that would make any emit in 1 suspect, so I think its still an acceptable change.

Reply:
Thanks for the detailed clarification. 👍🏻